Cmdline/Action/Replace.cs

```diff
@@ -0,0 +1,142 @@
+                                to_replace.Add(replacement);
+                            }
+                        }
+                        catch (ModuleNotFoundKraken)
+                        {
+                            log.InfoFormat("{0} is installed, but it or its replacement is not in the registry",
+                                mod.Key);
+                        }
+                    }
+                }
+            }
+            else
+            {
+                foreach (string mod in options.modules)
+                {
+                    try
+                    {
+                        log.DebugFormat("Checking that {0} is installed", mod);
+                        CkanModule modToReplace = registry.GetInstalledVersion(mod);
+                        if ( modToReplace != null) 
+                        {
+                            log.DebugFormat("Testing {0} {1} for possible replacement", modToReplace.identifier, modToReplace.version);
+                            try
+                            {
+                                // Check if replacement is available
+                                ModuleReplacement replacement = registry.GetReplacement(mod, ksp.VersionCriteria());
+                                if (replacement != null)
+                                {
+                                    // Replaceable
+                                    log.InfoFormat("Replacement {0} {1} found for {2} {3}",
+                                        replacement.ReplaceWith.identifier, replacement.ReplaceWith.version,
+                                        replacement.ToReplace.identifier, replacement.ToReplace.version);
+                                    to_replace.Add(replacement);
+                                }
+                                log.InfoFormat("Attempt to replace {0} failed, replacement {1} is not compatible",
+                                    mod, modToReplace.replaced_by.name);
+                            }
+                            catch (ModuleNotFoundKraken)
+                            {
+                                log.InfoFormat("{0} is installed, but its replacement {1} is not in the registry",
+                                    mod, modToReplace.replaced_by.name);
+                            }
+                        }
+                    }
+                    catch (ModuleNotFoundKraken kraken)
+                    {
+                        User.RaiseMessage("Module {0} not found", kraken.module);
+                    }                                           
+                }
+            }
+            // TODO: These instances all need to go.
+            ModuleInstaller.GetInstance(ksp, User).Replace(to_replace, new NetAsyncModulesDownloader(User));
+            User.RaiseMessage("\r\nDone!\r\n");
+
+            return Exit.OK;
+        }
+
+        internal static CkanModule LoadCkanFromFile(CKAN.KSP current_instance, string ckan_file)
```

Comment:
Let's share this function with `Install.cs` to avoid duplicating code. This file could call the other's, or vice versa, or this could go in a separate file.

Reply:
Action/Upgrade has another duplicate of this code. I don't know why it's internal

Reply:
`internal` is the same as public but only within the same project, so these classes could already share those functions as they are since they're all in the CmdLine project.